backend/app/models/interest.rb

```diff
@@ -27,15 +27,14 @@ def approve
       new_donor.name = donor_name
     end
 
-    Campaign.new(
+    self.campaign = Campaign.new(
       name: campaign_name,
       description: campaign_description,
       promised_amount: promised_amount,
       coupon_denomination: coupon_denomination,
       start: start,
       end: self.end,
       primary_donor: primary_donor,
-      interest: self,
       charities: charities
     )
```

Comment:
Note: `approve` method is broken after we added the compulsory `giving_sg_url`. It no longer makes sense to directly convert `Interest` to `Campaign` since we need to add the campaign url manually. KIV, out of scope of this MR.